ui/html-fragment.reel/html-fragment.js

```diff
@@ -0,0 +1,223 @@
+
+                                    if (shouldRemoveAttribute) {
+                                        child.removeAttribute(attributeName);
+                                        ll--;
+                                        ii--;
+                                    }
+                                } else {
+                                    delegateResponse = this.callDelegateMethod(
+                                        'htmlFragmentWillUseAttributeValueForAttributeNameFromNode',
+                                        this,
+                                        attributeValue,
+                                        attributeName,
+                                        child
+                                    );
+
+                                    if (typeof delegateResponse === 'string') {
+                                        attributeValue = delegateResponse;
+                                    }
+
+                                    child.setAttribute(attributeName, attributeValue);
+                                }
+                            }
+
+                            this._sanitizeNode(child, allowedTags, allowedAttributes);
+                        }
+                    }
+                }                
+            }
+
+            return parent;
+        }
+    },
+
+    sanitizeHtml: {
+        value: function (html, options) {
+            var doc;
+
+            if (window.DOMParser) {
+                try {
+                    doc = new DOMParser().parseFromString(html, "text/html");
+                } catch (DOMParserError) {
+                    // Ignore error
+                }
+
+                this._sanitizeNode(
+                    doc.body,
+                    options.allowedTags,
+                    options.allowedAttributes
+                );
+
+                this._sanitizeNode(
+                    doc.head,
+                    options.allowedTags,
+                    options.allowedAttributes
+                );
+            }
+
+            return doc;
+        }
+    },
+
+    draw: {
+        value: function () {
+            if (this.value && this.needsSanitizeHtml) {
+                var doc = this.sanitizeHtml(
+                    this.value, this._getSanitizerOptions()
+                );
+
+                if (doc) {
+                    var range = doc.createRange();
+                    range.selectNodeContents(doc.body);
+                    this.element.appendChild(range.extractContents());
+                    range.selectNodeContents(doc.head);
+                    document.head.appendChild(range.extractContents());
+                } else {
+                    this.element.innerHTML = '';
+                }                
+            } else {
+                this.element.innerHTML = '';
+            }
+
+            this.needsSanitizeHtml = false;
+        }
+    }
+  
+}, {
+    
+    DefaultSanitizerOptions: {
```

Comment:
might be worth exposing on prototype as well? -> defaultAllowedTags and defaultAllowedAttributes

Reply:
will do